engine/components/animation/AnimationController.cpp

```diff
@@ -1,12 +1,24 @@
 //
 // Created by 11896 on 19/11/2024.
 //
+
 #include "AnimationController.hpp"
 
-AnimationController::AnimationController(GameObject& gameObject, const char* tag, AnimationRenderable& renderable)
-		: Component(gameObject, tag), _renderable(renderable) {}
+#include "gameObject/GameObject.hpp"
+#include "components/renderables/AnimationRenderable.hpp"
+#include "logging/BLogger.hpp"
+
+AnimationController::AnimationController(GameObject* gameObject, const char* tag)
+		: Component(gameObject, tag), _renderable(nullptr) {}
 
-void AnimationController::Start() {}
+Component* AnimationController::_clone(const GameObject& parent) {
+	auto clone = new AnimationController(*this);
```

Comment:
Wordt deze new controller opgeruimd? Alsin we maken het aan met new, dus op heap, waarbij we uit moeten kijken of de Animationcontroller niet in memory blijft nadat de engine ermee kapt.